Backpack/src/main/java/net/skyscanner/backpack/snackbar/BpkSnackbar.kt

```diff
@@ -0,0 +1,130 @@
+import net.skyscanner.backpack.text.BpkText
+import net.skyscanner.backpack.util.BpkTheme
+import net.skyscanner.backpack.util.use
+
+/**
+ * A builder wrapper around [Snackbar] providing the required customization.
+ *
+ * [BpkSnackbar.builder] creates a new builder. It's strongly recommended to set text and action
+ * before you build a [Snackbar] using [BpkSnackbar.build] method.
+ *
+ * @see BpkSnackbar.builder
+ * @see BpkSnackbar.setText
+ * @see BpkSnackbar.setAction
+ * @see BpkSnackbar.setDuration
+ * @see BpkSnackbar.build
+ */
+class BpkSnackbar private constructor(
+  private val context: Context,
+  private val snackbar: Snackbar,
+  @ColorInt textColor: Int,
+  @ColorInt actionColor: Int,
+  @ColorInt backgroundColor: Int
+) {
+
+  private val textFontSpan = BpkFontSpan(context, BpkText.SM, BpkText.Weight.NORMAL)
+  private val textColorSpan = ForegroundColorSpan(textColor)
+
+  private val actionFontSpan = BpkFontSpan(context, BpkText.SM, BpkText.Weight.EMPHASIZED)
+  private val actionColorSpan = ForegroundColorSpan(actionColor)
+
+  init {
+    snackbar.setBackgroundColorCompat(backgroundColor)
+    snackbar.setMessageAppearanceCompat(textFontSpan, textColorSpan)
+    snackbar.setActionAppearanceCompat(actionFontSpan, actionColorSpan)
+  }
+
+  fun build(): Snackbar =
+    snackbar
+
+  fun setText(message: CharSequence): BpkSnackbar {
+    snackbar.setText(snackbar.customiseText(message, textFontSpan, textColorSpan))
+    return this
+  }
+
+  fun setText(@StringRes resId: Int): BpkSnackbar =
+    setText(context.getString(resId))
+
+  fun setAction(text: CharSequence, listener: View.OnClickListener): BpkSnackbar {
+    snackbar.setAction(snackbar.customiseText(text, actionFontSpan, actionColorSpan), listener)
+    return this
+  }
+
+  fun setAction(@StringRes resId: Int, listener: View.OnClickListener): BpkSnackbar =
+    setAction(context.getText(resId), listener)
+
+  inline fun setAction(text: CharSequence, crossinline listener: (View) -> Unit): BpkSnackbar =
+    setAction(text, View.OnClickListener { listener(it) })
+
+  inline fun setAction(@StringRes resId: Int, crossinline listener: (View) -> Unit): BpkSnackbar =
+    setAction(resId, View.OnClickListener { listener(it) })
+
+  fun setDuration(duration: Int): BpkSnackbar {
+    snackbar.duration = duration
+    return this
+  }
+
+  companion object {
+
+    @JvmStatic
+    val LENGTH_INDEFINITE = Snackbar.LENGTH_INDEFINITE
+
+    @JvmStatic
+    val LENGTH_SHORT = Snackbar.LENGTH_SHORT
+
+    @JvmStatic
+    val LENGTH_LONG = Snackbar.LENGTH_LONG
+
+    /**
+     * Creates a new builder for a [Snackbar] using given [text] and [duration].
+     * [view] provides theme and hierarchy to put the [Snackbar]
+     */
+    @SuppressLint("Recycle")
+    @JvmStatic
+    fun builder(view: View, text: CharSequence, duration: Int): BpkSnackbar {
+      val context = view.context
+
+      @ColorInt var textColor = BpkTheme.getColor(context, R.color.bpkWhite)
```

Comment:
Just a question, but what is the benefit of annotating a local variable that's never exposed?

Reply:
I've added this only for debugging as it provides some information to debugger. I can remove it if you want.